src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateUtility.cs

```diff
@@ -237,5 +237,21 @@ public static byte[] GetHash(X509Certificate2 certificate, HashAlgorithmName has
 
             return hashAlgorithm.ComputeHash(certificate.RawData);
         }
+
+        /// <summary>
+        /// Determines if a certificate is self-signed.
+        /// </summary>
+        /// <param name="certificate">The certificate to check.</param>
+        /// <returns>A boolean flag indicating if the certificate is self-signed.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="certificate" /> is <c>null</c>.</exception>
+        public static bool IsSelfSigned(X509Certificate2 certificate)
+        {
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            return certificate.IssuerName.RawData.SequenceEqual(certificate.SubjectName.RawData);
```

Comment:
https://security.stackexchange.com/a/162263

It seems like `issuer == subject name` is only one possible check we could do. Aren't issue name and subject name strings that could theoretically match for chain with more than one element?